### PR TITLE
*: small refactoring for metadata lock

### PR DESCRIPTION
**Pull request by wjhuang2016**
Signed-off-by: wjhuang2016 <huangwenjun1997@gmail.com>

### What problem does this PR solve?

ref https://github.com/pingcap/tidb/issues/37275

This PR does some preparation jobs for metadata lock.
The essential job is separating waitSchemaChanged from DDL worker.
After this PR, the general job maybe interleaving. That is,
DDL1 None, DDL2 None, DDL1 DeleteOnly, DDL2 DeleteOnly.
Before this PR: DDL1 None, DDL1 DeleteOnly, ... DDL1 Public, DDL2 None ...

Besides, replace `find_in_set` by `CONCAT(',', t2.table_ids, ',') REGEXP CONCAT(',', REPLACE(t1.table_ids, ',', '|'), ',')`
The later supports more than one ID.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [x] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: session/session.go [hunk 1]
```diff
@@ -1271,6 +1271,10 @@ func createSessionWithDomainFunc(store kv.Storage) func(*domain.Domain) (pools.R
 		if err != nil {
 			return nil, errors.Trace(err)
 		}
+		err = se.sessionVars.SetSystemVar(variable.MaxAllowedPacket, strconv.FormatUint(variable.DefMaxAllowedPacket, 10))
```
[the remark this answers is not in the document]
Comment (wjhuang2016) [reply session/session.go:1274]: Used in concat function

File: ddl/job_table.go [hunk 2]
```diff
@@ -239,8 +237,37 @@ func (d *ddl) delivery2worker(wk *worker, pool *workerPool, job *model.Job) {
...
+						mockDDLErrOnce = schemaVer
+						failpoint.Return()
+					}
+				}
+			})
+
+			// Here means the job enters another state (delete only, write only, public, etc...) or is cancelled.
+			// If the job is done or still running or rolling back, we will wait 2 * lease time to guarantee other servers to update
+			// the newest schema.
+			waitSchemaChanged(context.Background(), d.ddlCtx, d.lease*2, schemaVer, job)
```
[the remark this answers is not in the document]
Comment (wjhuang2016) [reply ddl/job_table.go:258]: It's ok not to exit the waitSchemaChanged when timeout. Using Background() is compatible with MDL.

File: session/session.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (tangenta) [reply session/session.go:1271]: Why do we need to set it to the default value explicitly? It should be the default value already.

File: ddl/util/util.go [hunk 3]
```diff
@@ -53,6 +53,8 @@ const (
 	DDLAllSchemaVersions = "/tidb/ddl/all_schema_versions"
 	// DDLGlobalSchemaVersion is the path on etcd that is used to store the latest schema versions.
 	DDLGlobalSchemaVersion = "/tidb/ddl/global_schema_version"
+	// SessionTTL is the etcd session's TTL in seconds.
+	SessionTTL = 90
```
[the remark this answers is not in the document]
Comment (wjhuang2016) [reply ddl/util/util.go:57]: Unify the session lease.

**Review by tangenta — APPROVED** [review]

**Review by xiongjiwei — APPROVED** [review]

File: ddl/ddl_worker.go [hunk 4]
```diff
@@ -1321,15 +1292,15 @@ func (w *worker) waitSchemaChanged(ctx context.Context, d *ddlCtx, waitTime time
 	// OwnerCheckAllVersions returns only when context is timeout(2 * lease) or all TiDB schemas are synced.
 	err = d.schemaSyncer.OwnerCheckAllVersions(ctx, latestSchemaVersion)
 	if err != nil {
-		logutil.Logger(w.logCtx).Info("[ddl] wait latest schema version to deadline", zap.Int64("ver", latestSchemaVersion), zap.Error(err))
+		logutil.Logger(d.ctx).Info("[ddl] wait latest schema version to deadline", zap.Int64("ver", latestSchemaVersion), zap.Error(err))
 		if terror.ErrorEqual(err, context.DeadlineExceeded) {
 			return
 		}
 		// Wait until timeout.
 		<-ctx.Done()
```
Comment (xiongjiwei) [inline ddl/ddl_worker.go:1300]: this will never be done

**Pull request merged**